formsOnline/formsOnline-war/src/main/java/org/silverpeas/components/formsonline/control/FormsOnlineSessionController.java

```diff
@@ -287,19 +321,158 @@ public RequestsByStatus getAllUserRequests() throws FormsOnlineDatabaseException
   }
 
   public RequestsByStatus getHomepageValidatorRequests() throws FormsOnlineDatabaseException {
-    return getService().getValidatorRequests(getComponentId(), isWorkgroupEnabled(), getUserId(),
+    return getService().getValidatorRequests(getRequestsFilter(), getUserId(),
         new PaginationPage(1, DEFAULT_ITEM_PER_PAGE));
   }
 
   public RequestsByStatus getAllValidatorRequests() throws FormsOnlineDatabaseException {
-    return getService()
-        .getValidatorRequests(getComponentId(), isWorkgroupEnabled(), getUserId(), null);
+    RequestsFilter filter = getRequestsFilter();
+    if (getCurrentForm() != null) {
+      filter.getFormIds().add(Integer.toString(getCurrentForm().getId()));
+    }
+    return getService().getValidatorRequests(filter, getUserId(), null);
+  }
+
+  public Form getCurrentEmptyForm() throws SilverpeasException {
+    PublicationTemplate pubTemplate = getCurrentPublicationTemplate(true);
+    return getEmptyForm(pubTemplate);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') "NullPointerException" will be thrown when invoking method "getEmptyForm()". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS2259)